clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerGroupMetadata.java

```diff
@@ -18,6 +18,10 @@
 
 import java.util.Optional;
 
+/**
+ * A metadata struct containing the consumer group information.
+ * Note: Any change to this class is considered public and requires a KIP.
+ */
```

Comment:
This is not introduced by this PR but: the fields can be declared final?

Reply:
Yes, they could